lang/cs/Org.Apache.REEF.Network/Group/Driver/Impl/GroupCommDriver.cs

```diff
@@ -99,9 +102,25 @@ public sealed class GroupCommDriver : IGroupCommDriver
         /// </summary>
         public string MasterTaskId { get; private set; }
 
+        /// <summary>
+        /// A communication group created using default group name and default number of tasks
+        /// It is created when this property is referenced first time.
+        /// </summary>
         public ICommunicationGroupDriver DefaultGroup
-        {
-            get { return _commGroups[_groupName]; }
+        {           
+            get
+            {
+                ICommunicationGroupDriver defaultGroup;
+                _commGroups.TryGetValue(_defaultGroupName, out defaultGroup);
```

Comment:
This statement needs to move inside the lock. Otherwise what if in this call defaultGroup turns out to be null but by the time the call enters the lock, someone else created the group.